src/generators/library.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as Generator from 'yeoman-generator';
+import yosay = require('yosay');
+import { exec } from 'shelljs';
+import replace = require('replace-in-file');
+import { PackageJson } from '../types';
+import { readJson } from '../util';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-unsafe-assignment
+const { version } = require('../../package.json');
+
+type Answers = {
+  name: string;
+  description: string;
+  org: string;
+  scope: string;
+};
+
+function containsInvalidChars(input: string): boolean {
+  const invalidChars = '!#$%^&*()12345679 ?/\\,.";\':|{}[]~`'.split('');
```

Comment:
@mdonnalley I believe numbers are allowed in both scope and name. I could not find a definitive set of rules. Did you find one?

Reply:
@peternhale I didn't find a definitive list of rules... I was operating under the assumption that generally speaking we would advise teams to avoid numbers in the scope and name. I know that there was a `b2b` plugin once a upon time that Claire disliked the naming of